### PR TITLE
updateLinkedThread route improvements, bug fix

### PR DESCRIPTION
**Pull request by gdjohnson**
Previously, an author_chain object was being passed to the backend, rather than an author_chain id string.

This branch also adds support for collaborators to link threads (in addition to admins, mods, and authors).

### CODE REVIEW

**Review by zakhap — APPROVED** [review]
Thank you for fixing this! Works on my machine as well

File: client/scripts/controllers/server/threads.ts [hunk 1]
```diff
@@ -553,7 +553,7 @@ class ThreadsController {
       linking_thread_id,
       linked_thread_id,
       address: app.user.activeAccount.address,
-      author_chain: app.user.activeAccount.chain,
+      author_chain: app.user.activeAccount.chain.id,
```
Comment (zakhap) [inline client/scripts/controllers/server/threads.ts:556]: it doesn't seem like `address` or `author_chain` ever get used in the route? let's remove them and then get this PR in.

**Review by zakhap — CHANGES_REQUESTED** [review]

File: client/scripts/controllers/server/threads.ts (hunk 1 above)
Comment (gdjohnson) [reply client/scripts/controllers/server/threads.ts:556]: Nah they get used for the checks. `lookupAddressIsOwnedByUser` requires it.

Actually, we should just standardize when we use the `lookup` helpers and what data they need. And maybe just pass it with req.user or someth.

**Review by zakhap — APPROVED** [review]
LGTM

**Pull request merged**